server/src/test/java/org/spine3/server/transport/GrpcContainerShould.java

```diff
@@ -66,6 +77,59 @@ public void start_server() throws IOException {
         verify(server).start();
     }
 
+    @SuppressWarnings("MagicNumber")
+    @Test
+    public void add_and_remove_parameters_form_builder() {
+        final GrpcContainer.Builder builder = GrpcContainer.newBuilder()
+                                                           .setPort(8080)
+                                                           .setPort(60);
+        assertEquals(60, builder.getPort());
+
+        int count = 3;
+        final List<ServerServiceDefinition> definitions = new ArrayList<>(count);
+
+        for (int i = 0; i < count; i++) {
+            final BindableService mockService = mock(BindableService.class);
+            final ServerServiceDefinition mockDefinition = ServerServiceDefinition
+                    .builder(String.format("service-%s", i))
+                    .build();
+            when(mockService.bindService()).thenReturn(mockDefinition);
+            definitions.add(mockDefinition);
+
+            builder.addService(mockService);
+        }
+
+        count--;
+        builder.removeService(definitions.get(count));
+
+        final Set<ServerServiceDefinition> serviceSet = builder.getServices();
+        assertSize(count, serviceSet);
+
+        final GrpcContainer container = builder.build();
+        assertNotNull(container);
+    }
+
+    @Test
+    public void add_shutdown_hook_to_runtime() throws NoSuchFieldException, IllegalAccessException, IOException {
```

Comment:
`public void stop_properly_upon_application_shutdown`.

In tests you tell what should happen, not how it is implemented.